packages/lit-dev-tests/src/playwright/playground.spec.ts

```diff
@@ -70,16 +79,90 @@ test.describe('Playground', () => {
     ).toMatchSnapshot('helloWorldPlaygroundProject.png');
   });
 
-  test('open and close share menu', async ({page}) => {
+  test('share long url', async ({page}) => {
     await page.goto('/playground/?mods=gists');
+    await freezeSnackbars(page);
+
+    // Type some new content
+    await page.click('playground-code-editor');
+    await page.keyboard.press('Control+A');
+    await page.keyboard.type('"my long url content";');
     await waitForPlaygroundPreviewToLoad(page);
+
+    // Open the share menu
     await page.click('litdev-playground-share-button');
     await expect(await page.screenshot()).toMatchSnapshot(
-      'openAndCloseShareMenu-1-opened.png'
+      'shareLongUrl-1-shareMenuOpen.png'
+    );
+
+    // Save the long URL
+    await page.click('litdev-playground-share-long-url copy-button');
+    await page.waitForURL(/#project=/);
+    expect(await page.evaluate(() => navigator.clipboard.readText())).toMatch(
+      page.url()
+    );
+    await page.waitForSelector('litdev-playground-share-button litdev-flyout', {
+      state: 'hidden',
+    });
+    await expect(await page.screenshot()).toMatchSnapshot(
+      'shareLongUrl-2-snackbarOpen.png'
+    );
+
+    // Reload the page to confirm the new content is still there
+    await page.reload();
+    await waitForPlaygroundPreviewToLoad(page);
+    await expect(await page.screenshot()).toMatchSnapshot(
+      'shareLongUrl-3-pageReloaded.png'
     );
+  });
+
+  test('share gist', async ({page}) => {
+    await page.goto('/playground/?mods=gists');
+    await freezeSnackbars(page);
+
+    // Type some new content
     await page.click('playground-code-editor');
+    await page.keyboard.press('Control+A');
+    await page.keyboard.type('"my gist content";');
+    await waitForPlaygroundPreviewToLoad(page);
+
+    // Open the share menu
+    await page.click('litdev-playground-share-button');
+    await expect(await page.screenshot()).toMatchSnapshot(
+      'shareGist-1-shareMenuOpen.png'
+    );
+
+    // Sign in to GitHub
+    const [popup] = await Promise.all([
+      page.waitForEvent('popup'),
+      page.click('#signInButton'),
+    ]);
+    await popup.waitForLoadState();
+    await popup.click('text=Authorize lit');
```

Comment:
The fact that you have captured the entire flow in this change is magical.
Could we also capture the popup - or waiting till it is more fleshed out?

Reply:
You mean a screenshot of the popup? I was thinking there there isn't much point, since it's just a fake, and if we managed to click the Authorize button, that's good enough. And in reality, the design of the popup is out of our control so again maybe not too useful?

Reply:
Yep that makes sense!